Adding instructions and discussion for folders (#2045)
docker-mailserver/docker-mailserver: docs/content/examples/uses-cases/folders.md

---
title: 'Use Cases | Folders'
hide:
  - toc # Hide Table of Contents for this page
---
## Introduction

> This provides a solution letting different email clients sharing the same archive folder.

Reviewer: Does this really mean "share" the same folder (between accounts) or "share" the same configuration?

This wording might be confusing depending on what is meant indeed.

> It might also help when troubleshooting sharing `Sent` and `Drafts` folders. 

Reviewer: Same here

## Adding a mailbox folder

See [`target/dovecot/15-mailboxes.conf`][gh-config-dovecot-mailboxes] for existing folder definitions you can modify or enable.
Dovecot also [provides their own example config][dovecot-config-mailboxes] for reference, with further information in [their documentation][dovecot-docs-mailboxes].

## Caution

### Adding folders to an existing setup

Handling of newly added mailbox folders can be inconsistent across mail clients:

- Users may experience issues such as archived emails only being available locally.
- Users may need to migrate emails manually between two folders.

### Special Use Mailbox Support

Not all mail clients support Special Use Mailbox (_defined in [RFC 6154][rfc-6154]_), using the exact mailbox name instead. To support those clients, you may prefer to keep the tag and folder name identical.

### Internationalization (i18n)

> Users and mail clients may prefer localized mailbox names instead of English. Take care to test localized names work well, keep in mind concerns such as tag support.

Reviewer 2: As you are making commits and resolving prior review with no discussion, "tag" is presumably still unclear? Please correct the naming or provide context what a "tag" refers to in the documentation.

I am placing trust in your knowledge on the topic or other reviewers here. My only involvement is QA for docs.

Author: @polarathene I think I removed the reference of "tag" everywhere. And now it should only use "Special Use Mailbox" which is the term in RFC. I saw this "tag" usage when I search for the solution for this issue. But yes, it may cause confusion.

Reviewer 2: "tag" is still referenced in the current document:

> _To support those clients, you may prefer to keep the tag and folder name identical._

> _keep in mind concerns such as tag support._

If "tag" is a common way to talk about it, like "folder" for mailbox, then just hint that to the reader. eg: `### Special Use Mailbox (_aka Tag_) Support`?

If you do that, then it is clear what "tag" refers to in the quotes to someone who may not know any of this from anywhere else, just reading the example page you are contributing and interested in the feature.

Author: Updated PR with "tags" references removed. It's much less common than "folder".

> ### Email Clients Support
> *	Thunderbird
> 
> 	If new email account is added without `Special Use Mailbox` enabled for archives, Thunderbird suggests and may create `Archives` folder on server.
> 	If new email account is added after `Special Use Mailbox` enabled for archives, it will pick up the name assigned.
> 
> *	Outlook for Android
> 	
> 	If new email account is added without `Special Use Mailbox` enabled for archive, it archives locally.
> 	If new email account is added after `Special Use Mailbox` enabled for archives, it will pick up the name assgined.
> 
> *	Spark for Android
> 	
> 	If new email account is added without `Special Use Mailbox` enabled for archive, it archives on server folder named `Archive`.
> 	If new email account is added after `Special Use Mailbox` enabled for archives, it will pick up the name assgined.
> 
> * 	Windows Mail
> 	
> 	People suggested that it will look for the name instead of `Special Use Mailbox`.

Reviewer 2: ```suggestion
### Email Clients Support

#### Thunderbird

- If a new email account is added without `Special Use Mailbox` enabled for archives, Thunderbird suggests and may create an `Archives` folder on the server.
- If a new email account is added after `Special Use Mailbox` is enabled for archives, it will pick up the name assigned.

#### Outlook for Android

- If a new email account is added without `Special Use Mailbox` enabled for archives, it archives locally.
- If a new email account is added after `Special Use Mailbox` is enabled for archives, it will pick up the name assigned.

#### Spark for Android

- If new email account is added without `Special Use Mailbox` enabled for archives, it archives on the server folder named `Archive`.
- If new email account is added after `Special Use Mailbox` is enabled for archives, it will pick up the name assigned.

#### Windows Mail

- People suggested that it will look for the name instead of `Special Use Mailbox`.
```

Why are there tabs used here? I suggest changing the style to sub-headings with bullet points added.

Alternatively, since there is repetition, this may be better:

```suggestion
### Email Clients Support

- If a new email account is added without `Special Use Mailbox` enabled for archives:
    - Thunderbird suggests and may create an `Archives` folder on the server.
    - _Outlook for Android_ archives locally.
    - _Spark for Android_ archives on the server folder named `Archive`.
- If a new email account is added after `Special Use Mailbox` is enabled for archives:
    - _Thunderbird_, _Outlook for Android_ and _Spark for Android_ will pick up the name assigned.

!!! caution "Windows Mail"

    _Windows Mail_ has been said to ignore `Special Use Mailbox` and look only at the name assigned.
```

I think this is better? I corrected grammar a bit and made some slight modifications, I hope it is still accurate.

It would be even better if there was a link to any resources for each statement, but it is not required for approval.

Author: I wanted to highlight behavior per email client. And also I was trying to tell readers that what would happen if they create this folder later. But since you are also a reader of this page, if you think this is more clear, I am fine with it.

Reviewer 2: The 2nd suggestion just moves the email client names into the common conditions you repeated. The names are all still there and can be found with `ctrl+f` or reading the start of the lines that they apply to.

If italic style for the mail client names is not as good as bold, you could try that?

I think that the reader is someone who is interested in providing this feature with `docker-mailserver` to their users and wants to know the behaviour for different e-mail clients, but doesn't have any specific one in mind? If so reducing the repetition with the 2nd suggested version is better.
